docker-entrypoint.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#######################################
+# Enables jemalloc as the default memory allocator if it's available, otherwise the default glibc
+# allocator is used. jemalloc is added to avoid the memory leak created by the glibc allocator.
+# Refer https://github.com/google/fhir-data-pipes/issues/777 for details.
+#######################################
+enable_jemalloc() {
```

Comment:
Two points about this function:
1) If this is copied from Flink repo, please make a reference to it.
2) Do we need this kind of fallback logic? I mean since we are using this for the specific/known case of  `eclipse-temurin:17-jdk-focal` we should know where jemalloc will be installed, right?

Reply:
1. Reference to the Flink repo is added.
2. The fallback logic is required because the installation path of jemalloc is dependent on the processor architecture where the docker is installed i.e. if the docker is hosted by a macOS with arm64 processor architecture then the jemalloc installation path in the Docker installed is `/usr/lib/aarch64-linux-gnu/libjemalloc.so`.

Reply:
I see makes sense; thanks.